src/plugins/plugin.legend.js

```diff
@@ -411,15 +412,30 @@ var Legend = Element.extend({
 			// Horizontal
 			var isHorizontal = me.isHorizontal();
 			if (isHorizontal) {
+				// default to center as this is the original behaviour of just position
+				var alignedX = ((legendWidth - lineWidths[0]) / 2) + labelOpts.padding;
+				if (opts.alignment === 'left') {
+					alignedX = 0;
+				} else if (opts.alignment === 'right') {
+					alignedX = ((legendWidth - lineWidths[0])) + labelOpts.padding;
+				}
 				cursor = {
-					x: me.left + ((legendWidth - lineWidths[0]) / 2) + labelOpts.padding,
+					x: me.left + alignedX,
 					y: me.top + labelOpts.padding,
 					line: 0
 				};
 			} else {
+				// default to top as this is the original behaviour of just position
+				var alignedY = me.top + labelOpts.padding;
+				var legendHeight = (fontSize + labelOpts.padding) * me.legendItems.length;
```

Comment:
A legend may have multiple columns, so this calculation is not always correct. The number of items per column is smaller one of `me.legendItems.length` or the maximum number of items that can fit in `me.minSize.height`.
```javascript
var maxItemsPerColumn = Math.floor((me.minSize.height - labelOpts.padding) / itemHeight);
var legendHeight = Math.min(me.legendItems.length, maxItemsPerColumn) * itemHeight + labelOpts.padding;
```